sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -2555,6 +2555,18 @@ class DataFrameSuite extends QueryTest
     val df = Seq(0.0 -> -0.0).toDF("pos", "neg")
     checkAnswer(df.select($"pos" > $"neg"), Row(false))
   }
+
+  test("SPARK-32635: Replace references with foldables coming only from the node's children") {
```

Comment:
Although this end-to-end test is also useful, we had better have a narrow-downed test coverage at `FoldablePropagationSuite` in `catalyst` module. Could you add one please? If then, we can easily detect regression in `catalyst` module test instead of a full `sql` testing.

Reply:
Ok, I will try to add a test case to `FoldablePropagationSuite` tomorrow.

Reply:
Thanks!